### ISSUE TITLE
Sketch out a design for the monorepo structure

### ISSUE THREAD

**Issue by dglazkov** [polymath-ai/architecture#1]

**Comment by dglazkov** [comment]
I am thinking something like a `core` monorepo that contains most of the things, and then a bunch of separate repos for "applied Polymath": a firebase-configured thingy, a remix-configured thingy, etc.

**Comment by dglazkov** [comment]
On the other hand, maybe all of them should be in there: why keep some separate?

**Comment by dglazkov** [comment]
What's a good rubric here? Why would I keep a project outside of a monorepo?

**Comment by dalmaer** [comment]
I could see experimental stuff outside and then bring it in when it's blessed?

**Comment by dglazkov** [comment]
sounds good. Let's figure out how the dependencies will work. For the hypotheticl structure:
```
polymath-ai/core -- main repo where all the cool stuff happens
polymath-ai/core/packages/cli -- the cli
polymath-ai/core/packages/client -- the client library
```
Can `cli` depend on `client`?

**Comment by dglazkov** [comment]
https://turbo.build/repo/docs/handbook/sharing-code#anatomy-of-a-package looks like it.

Let me play with this a bit.

**Comment by dglazkov** [comment]
it can!

**Comment by dglazkov** [comment]
Sounds like `changeset` is an important part of the workflow.

**Comment by dglazkov** [comment]
I just realized that instead of generic `packages`, we can use multiple dirs and signal the stage of the package through them. I am thinking three pace layers:

- wild experiments -- the fastest-moving, most likely to fails and be abandoned
- useful bits that we see as part of the overall project -- things that are leaf nodes in the project, like remix or firebase starter projects.
- what really makes up the project -- things like `client` and `cli`.

**Comment by dglazkov** [comment]
Naming suggestion:

- `seeds`
- `leafs`
- `core` or `stem`

**Comment by dglazkov** [comment]
More boring:

- `experiments`
- `kit`
- `core`

**Comment by dalmaer** [comment]
I like core and experiments.

Do items like `architecture` fit in core? Or it's own thing?

Where does a `kit` fit in? Do we need 3?

**Comment by dglazkov** [comment]
Here's what I am thinking, looking at current repos:

format: `[monorepo dir]` <-- from `[polymath-ai repo]`

`/core/cli`
`/core/client`
`/core/config` <-- from `configuration`
`/kits/remix-starter` <-- from `polymath-remix`
`/kits/firebase-starter` <-- from `firebase-fun`
`/kits/cloudflare-starter` <-- from `cloudflaring-fun`.

WDYT?

**Comment by dglazkov** [comment]
`architecture` stays out of the monorepo, since it's not Node-specific.

**Comment by dalmaer** [comment]
Ah,so the mono repo is for all things node (or even js in case at some point we ditch the node apis).

What do you think about having one repo that's got all of the docs and decisions for the protocol / spec / vision etc that's neutral to impl. A bunch of this is in original polymath because that was the only spot.

Then we have a monorepo for the js world, one for python, and room for other potentially (maybe just a client in rust say... no time soon...).

And now I get kit. A kit implements full web endpoints that you can grab to get started faster with.

**Comment by dglazkov** [comment]
Yes! The one with all the docs and decisions -- is the `architecture`. Does that make sense? I already migrated https://github.com/polymath-ai/architecture/blob/main/format.md here.

**Comment by dglazkov** [comment]
I got some of the repos wrong. Let me do this again:


`/core/cli`
`/core/client`
`/core/config` <-- from `configuration`
`/kits/remix` <-- from `polymath-remix`
`/experimental/firebase` <-- from `firebase-fun`
`/experimental/cloudflare` <-- from `cloudflaring-fun`.

**Comment by dglazkov** [comment]
Maybe like `seeds` instead of `experimental` -- make it more playful and with fewer letters

**Comment by dalmaer** [comment]
Perfect. I will move over GLOSSARY and the poem etc lol
